src/op_analytics/cli/subcommands/pulls/agora/delegates.py

```diff
@@ -0,0 +1,232 @@
+        offsets = list(range(0, self.max_offset, self.limit))[:10]
+        data = run_concurrently(
+            function=lambda offset: self.request(offset).data,
+            targets=offsets,
+            max_workers=self.max_workers,
+        )
+        return list(itertools.chain.from_iterable(data.values()))
+
+
+def pull_delegates():
+    """Pull and write agora delegates data."""
+    p = Paginator(url=f"{BASE_URL}/delegates", params={"sort": "voting_power"}, max_workers=32)
+    delegate_data = p.concurrent_fetch()
+
+    # Clean and transform to Pandas
+    normalized_data = []
+    for item in delegate_data:
+        # Extract fields from the item dictionary
+        address = item.get("address", "")
+        voting_power = item.get("votingPower", {})
+        citizen = item.get("citizen", False)
+        statement = item.get("statement", {})
+
+        # Make sure voting_power is a dictionary
+        if not isinstance(voting_power, dict):
+            continue
+
+        flattened = {
+            "address": address,
+            "voting_power_total": voting_power.get("total", "0"),
+            "voting_power_direct": voting_power.get("direct", "0"),
+            "voting_power_advanced": voting_power.get("advanced", "0"),
+            "is_citizen": citizen,
+        }
+
+        if isinstance(statement, dict):
+            flattened.update(
+                {
+                    "statement_signature": statement.get("signature", ""),
+                    "statement_created_at": statement.get("created_at", ""),
+                    "statement_updated_at": statement.get("updated_at", ""),
+                    "statement_text": statement.get("payload", {}).get("delegateStatement", ""),
+                    "twitter": statement.get("payload", {}).get("twitter", ""),
+                    "discord": statement.get("payload", {}).get("discord", ""),
+                }
+            )
+
+        normalized_data.append(flattened)
+    df = pd.DataFrame(normalized_data)
+    df = df.astype(
+        {
+            "address": "string",
+            "voting_power_total": "Float64",
+            "voting_power_direct": "Float64",
+            "voting_power_advanced": "Float64",
+            "is_citizen": "bool",
+            "statement_signature": "string",
+            "statement_created_at": "string",
+            "statement_updated_at": "string",
+            "statement_text": "string",
+            "twitter": "string",
+            "discord": "string",
+        },
+        errors="coerce",
+    )
+    # Check for duplicates
+    if df.shape[0] != len(df["address"].drop_duplicates()):
+        df = df.drop_duplicates()
+        log.warning(
+            f"Found duplicates in delegates data. Dropped {df.shape[0] - len(df['address'].drop_duplicates())} rows."
+        )
+
+    # Todo: This is to partition by pull date. Figure out if this is the correct way to do this.
+    df["dt"] = now_dt()
+
+    # Convert to polars
```

Comment:
Maybe we should stay in polars world from the start?  Given that it seems we are not using any functionality that is very specific to pandas.